shared/src/commonTest/kotlin/com/mbta/tid/mbta_app/model/RouteCardDataTest.kt

```diff
@@ -638,10 +662,1577 @@ class RouteCardDataTest {
+
+        objects.schedule {
+            trip = trip1
+            stopId = stop.id
+            stopSequence = 90
+            departureTime = time - 2.hours
+        }
+        val staticData =
+            NearbyStaticData.build {
+                route(route) {
+                    stop(stop) {
+                        headsign("A", listOf(routePatternA))
+                        headsign("B", listOf(routePatternB))
+                        headsign("C", listOf(routePatternC))
+                    }
+                }
+            }
+
+        assertEquals(
+            listOf(
+                StopsAssociated.WithRoute(
+                    route,
+                    listOf(
+                        PatternsByStop(
+                            route,
+                            stop,
+                            listOf(
+                                RealtimePatterns.ByHeadsign(
+                                    route,
+                                    "A",
+                                    null,
+                                    listOf(routePatternA),
+                                    emptyList()
+                                ),
+                                RealtimePatterns.ByHeadsign(
+                                    route,
+                                    "B",
+                                    null,
+                                    listOf(routePatternB),
+                                    emptyList(),
+                                    hasSchedulesToday = false
+                                )
+                            )
+                        )
+                    )
+                )
+            ),
+            staticData.withRealtimeInfo(
+                globalData = GlobalResponse(objects),
+                sortByDistanceFrom = stop.position,
+                schedules = ScheduleResponse(objects),
+                predictions = PredictionsStreamDataResponse(objects),
+                alerts = AlertsStreamDataResponse(emptyMap()),
+                filterAtTime = time,
+                pinnedRoutes = setOf(),)
+        )
+    }
+
+    @Test
+    fun `RouteCardData routeCardsForStopList checks route along with route pattern and stop`()  {
+        val objects = ObjectCollectionBuilder()
+        val stop = objects.stop()
+        val route1 = objects.route { sortOrder = 1 }
+        val routePattern1 = objects.routePattern(route1) { representativeTrip { headsign = "A" } }
+        val trip1 = objects.trip(routePattern1)
+
+        val route2 = objects.route { sortOrder = 2 }
+        val routePattern2 = objects.routePattern(route2) { representativeTrip { headsign = "B" } }
+        val trip2 = objects.trip(routePattern2)
+
+        // change from before: this will be included because it has a matching
+        // route & direction, even if it is an unexpected route pattern.
+        val trip3 = objects.trip(routePattern2) { routePatternId = "not the right id" }
```

Comment:
This is a change from `NearbyStaticData.withRealtimeInfo` that seems desirable based on the theory behind the initial trip headsigns work - if there is a trip in a known route + direction, we should show that trip.